### PR TITLE
fix for ticket #293

### PR DESCRIPTION
**Pull request by M-a-x-G**

### CODE REVIEW

File: spoon-client/src/main/java/com/squareup/spoon/Spoon.java [hunk 1]
```diff
@@ -229,19 +229,34 @@ private static File filesDirectory(Context context, String directoryType, String
   static StackTraceElement findTestClassTraceElement(StackTraceElement[] trace) {
     for (int i = trace.length - 1; i >= 0; i--) {
       StackTraceElement element = trace[i];
-
+      int apiVersion = Build.VERSION.SDK_INT;
       if (TEST_CASE_CLASS_JUNIT_3.equals(element.getClassName()) //
           && TEST_CASE_METHOD_JUNIT_3.equals(element.getMethodName())) {
-        return trace[i - 3];
+          //Stacktrace length changed in M
```
Comment (edenman) [inline spoon-client/src/main/java/com/squareup/spoon/Spoon.java]: lots of dup code.  can you pull this into a helper method called `extractStackElement` or somesuch?

**Comment by edenman** [comment]
Also @M-a-x-G you'll need to sign our [CLA] (https://docs.google.com/a/squareup.com/forms/d/13WR8m5uZ2nAkJH41k7GdVBXAAbzDk00vxtEYjd6Imzg/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ) for me to merge this.

**Comment by M-a-x-G** [comment]
I signed the CLA a second time. Is there still something wrong?

**Comment by edenman** [comment]
Sorry, I don't get notified when the CLA is signed.  Can you squash this into a single commit and then comment again to let me know?

**Comment by JakeWharton** [comment]
Take the squashing opportunity to create a more descriptive commit message, too! Ideally there's no references to issue numbers and the first sentence describes the behavior change and motivation. Maybe like "Account for different stack trace elements on newer versions of Android".

**Comment by edenman** [comment]
Manually squashed and merged.  Thanks, @M-a-x-G!

**Pull request closed without merging**